### PR TITLE
Fix possibly wrong CFG_* values in cross-compilation scenarios

### LINKED ISSUE #36: CFG_TARGET_ARCH has wrong value when cross-compiling
When I build via `cargo build --target=aarch64-apple-darwin` with the `x86_64-apple-darwin` toolchain, `CFG_TARGET_ARCH` contains `x86_64` whereas it should contain `aarch64`. Probably also affects other constants and other cross compilation scenarios.

I think this is due to `built`'s use of the `cfg!` macro which returns the architecture that `build.rs` is built with instead of the architecture that the crate is built with. See [here](https://github.com/RustCrypto/asm-hashes/issues/15).

Therefore I guess it would be more correct if `built` would use the `CARGO_CFG_*` environment variables (see [here](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts)). And also easier to implement because you don't have to make multiple `cfg!()` calls until you find the correct one.

### PR DESCRIPTION
**Pull request by helgoboss**
Should fix #36.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -638,59 +638,56 @@ fn write_time(w: &mut fs::File) -> io::Result<()> {
...
+    let target_family = get_env("CARGO_CFG_TARGET_FAMILY");
+    let target_os = get_env("CARGO_CFG_TARGET_OS");
+    let target_pointer_width = get_env("CARGO_CFG_TARGET_POINTER_WIDTH");
 
     write_str_variable!(
         w,
         "CFG_TARGET_ARCH",
-        target_arch(),
+        target_arch,
         "The target architecture, given by `cfg!(target_arch)`."
```
Comment (lukaslueg) [inline src/lib.rs:656]: The docs should reflect this change, something to the tune "given by CARGO_CFG_TARGET_ARCH"

**Comment by lukaslueg** [comment]
Thanks for the PR, I think the change is valid, although I mildly wonder about backwards compatibility. The doc-strings, the module docs and the README have to reflect this change, as `cfg!`-values are no longer used at all.

Can we have a test for this? At least on Travis we can cross-compile to `x86` and make sure that it's picked up - just to prevent regressions of sorts.

File: src/lib.rs (hunk 1 above)
Comment (helgoboss) [reply src/lib.rs]: Okay, adjusted.

**Comment by helgoboss** [comment]
> Thanks for the PR, I think the change is valid, although I mildly wonder about backwards compatibility. The doc-strings, the module docs and the README have to reflect this change, as `cfg!`-values are no longer used at all.

Yes, it's non-backward compatible because the results will change for cross compilation builds. Probably justifies at least a minor version bump.

> Can we have a test for this? At least on Travis we can cross-compile to `x86` and make sure that it's picked up - just to prevent regressions of sorts.

Would be nice. Have no time though to look into this though at the moment.

**Comment by lukaslueg** [comment]
CI seems legit. Interestingly enough, `cfg!` for `env` will return `gnu` on `darwin`, but `CARGO_CFG_TARGET_ENV` is the empty string (and documented as such). Could you change

https://github.com/lukaslueg/built/blob/226a16a5b952735887acb4f6909414add769094c/tests/testbox_tests.rs#L221

to `let _: &'static str = built_info::CFG_ENV` or similiar to work around this? This ok then if CI is green

**Comment by helgoboss** [comment]
Is it correct that it results in an empty string in this case?

**Comment by lukaslueg** [comment]
No, it seems `env` remains undefined on MacOS, but executing the test on musl or similar should yield something. We can't really test for that universally (without running circles), which is why all the tests only assert that those variables have _some_ value. For for `env` we can only test that it can be assigned to a `&'static str` after this PR.

**Comment by lukaslueg** [comment]
Thanks!

**Comment by lukaslueg** [comment]
I pushed this and other changes as `0.5.0`

**Pull request merged**